File: ruby/new_task.rb
#!/usr/bin/env ruby
require 'bunny'

# create connection
conn = Bunny.new(automatically_recover: true)
conn.start

# create channel
ch = conn.create_channel

# define queue
q = ch.queue('task_queue', durable: true)

# get message
msg = ARGV.empty? ? 'Hello from Ruby task...' : ARGV.join(' ')

# send a message
q.publish(msg, persistent: true)
puts " [x] Sent #{msg}"

# close a connection
conn.close

File: elixir/new_task.exs
# create connection
{:ok, connection} = AMQP.Connection.open
{:ok, channel} = AMQP.Channel.open(connection)

# create channel
AMQP.Queue.declare(channel, "task_queue", durable: true)

# get message
message =
  case System.argv do
    []    -> "Hello from Elixir task..."
    words -> Enum.join(words, " ")
  end

# send message
AMQP.Basic.publish(channel, "", "task_queue", message, persistent: true)
IO.puts " [x] Send '#{message}'"

AMQP.Connection.close(connection)
